packages/ckeditor5-widget/tests/widgettypearound/widgettypearound.js

```diff
@@ -0,0 +1,345 @@
+/**
+ * @license Copyright (c) 2003-2020, CKSource - Frederico Knabben. All rights reserved.
+ * For licensing, see LICENSE.md or https://ckeditor.com/legal/ckeditor-oss-license
+ */
+
+import ClassicEditor from '@ckeditor/ckeditor5-editor-classic/src/classiceditor';
+import ArticlePluginSet from '@ckeditor/ckeditor5-core/tests/_utils/articlepluginset';
+import Paragraph from '@ckeditor/ckeditor5-paragraph/src/paragraph';
+import DomEventData from '@ckeditor/ckeditor5-engine/src/view/observer/domeventdata';
+import EventInfo from '@ckeditor/ckeditor5-utils/src/eventinfo';
+import global from '@ckeditor/ckeditor5-utils/src/dom/global';
+
+import Widget from '../../src/widget';
+import WidgetTypeAround from '../../src/widgettypearound/widgettypearound';
+import { toWidget } from '../../src/utils';
+
+import { setData as setModelData, getData as getModelData } from '@ckeditor/ckeditor5-engine/src/dev-utils/model';
+
+describe( 'WidgetTypeAround', () => {
+	let element, plugin, editor, editingView, viewDocument, viewRoot;
+
+	beforeEach( async () => {
+		element = global.document.createElement( 'div' );
+		global.document.body.appendChild( element );
+
+		editor = await ClassicEditor.create( element, {
+			plugins: [
+				ArticlePluginSet, Widget,
+
+				blockWidgetPlugin, inlineWidgetPlugin
+			]
+		} );
+
+		editingView = editor.editing.view;
+		viewDocument = editingView.document;
+		viewRoot = viewDocument.getRoot();
+		plugin = editor.plugins.get( WidgetTypeAround );
+	} );
+
+	afterEach( async () => {
+		element.remove();
+
+		await editor.destroy();
+	} );
+
+	describe( 'plugin', () => {
+		it( 'is loaded', () => {
+			expect( editor.plugins.get( WidgetTypeAround ) ).to.be.instanceOf( WidgetTypeAround );
+		} );
+
+		it( 'requires the Paragraph plugin', () => {
+			expect( WidgetTypeAround.requires ).to.deep.equal( [ Paragraph ] );
+		} );
+	} );
+
+	describe( '_insertParagraph()', () => {
```

Comment:
`_insertParagraph` should be `protected` (not `private`) to make it accessible for tests.